daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/ZonedTextParsers.scala

```diff
@@ -95,17 +95,13 @@ case class ConvertZonedNumberParser(
         return
       }
 
-      val numValue = {
-        if (!primNumeric.isValidRange(num)) {
-          PE(start, "Parsed %s is out of range for type: %s",
-            context.optPrimType.get.globalQName, str, num)
+      val numValue = try {
+        primNumeric.fromNumber(num)
+      } catch {
+        case e: Throwable => {
+          PE(start, e.getMessage)
           return
         }
-
-        // convert to proper type
-        val asNumber = primNumeric.fromNumber(num)
-
-        asNumber
```

Comment:
These is a great refactoring! Gets rid of all the random places we do range checking and moves it into a single place in NodeInfo. Good addition.